src/libpriv/rpmostree-rpm-util.c

```diff
@@ -1057,3 +1057,25 @@ rpmostree_fcap_to_xattr_variant (const char *fcap)
                                                    vfsbytes, FALSE));
   return g_variant_ref_sink (g_variant_builder_end (&builder));
 }
+
+gboolean
+rpmostree_get_pkg_chksum_repr (DnfPackage *pkg,
+                               char      **out_chksum_repr,
+                               GError    **error)
+{
+  int chksum_type;
+  g_autofree char *chksum = NULL;
+  const unsigned char *chksum_raw = NULL;
+
+  chksum_raw = dnf_package_get_chksum (pkg, &chksum_type);
```

Comment:
Yeah, I briefly looked into this. Ideally we'd store every checksum available and only require one match, but I couldn't see how to do this easily with libsolv, if at all (not to mention we'd have to teach it to libdnf first).